src/NuGet.Core/NuGet.Protocol.Core.Types/SourceCacheContext.cs

```diff
@@ -95,18 +79,41 @@ private TimeSpan GetCacheTime(DateTimeOffset? maxAge, TimeSpan defaultTime)
             return timeSpan;
         }
 
-        public string GeneratedTempFolder { get; } = Path.Combine(Path.GetTempPath(), "NuGet", "TempCache", _tempFolderGuid);
+        public string GeneratedTempFolder
+        {
+            get
+            {
+                if (_generatedTempFolder == null)
+                {
+                    var newTempFolder = Path.Combine(
+                        Path.GetTempPath(),
+                        "NuGet",
+                        "TempCache",
+                        Guid.NewGuid().ToString());
+
+                    Interlocked.CompareExchange(ref _generatedTempFolder, newTempFolder, null);
```

Comment:
I would expect this path to be set in the constructor the same as before, just with a non-static guid.